src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/NodeService.kt

```diff
@@ -223,4 +119,134 @@ class NodeService(
       is TransactionWithClosureError -> getPaymentTypeCode(tx.transactionAtPreviousState)
       else -> TIPO_VERSAMENTO_CP
     }
+
+  private fun buildClosePaymentForCancellationRequest(
+    transactionWithCancellation: BaseTransactionWithCancellationRequested,
+    transactionId: TransactionId
+  ): ClosePaymentRequestV2Dto {
+    val amount =
+      EuroUtils.euroCentsToEuro(
+        transactionWithCancellation.paymentNotices
+          .stream()
+          .mapToInt { el -> el.transactionAmount.value }
+          .sum())
+    return ClosePaymentRequestV2Dto().apply {
+      paymentTokens = transactionWithCancellation.paymentNotices.map { el -> el.paymentToken.value }
+      outcome = ClosePaymentRequestV2Dto.OutcomeEnum.KO
+      this.transactionId = transactionId.value()
+      transactionDetails =
+        TransactionDetailsDto().apply {
+          transaction =
+            TransactionDto().apply {
+              this.transactionId = transactionId.value()
+              this.transactionStatus = getTransactionDetailsStatus(transactionWithCancellation)
+              this.creationDate = transactionWithCancellation.creationDate.toOffsetDateTime()
+              this.amount = amount
+              this.grandTotal = amount
+            }
+          info =
+            InfoDto().apply {
+              type = getPaymentTypeCode(transactionWithCancellation)
+              clientId = transactionWithCancellation.clientId.name
+            }
+          user = UserDto().apply { type = UserDto.TypeEnum.GUEST }
+        }
+    }
+  }
+
+  private fun buildAuthorizationCompletedClosePaymentRequest(
+    authCompleted: BaseTransactionWithCompletedAuthorization,
+    transactionOutcome: ClosePaymentRequestV2Dto.OutcomeEnum,
+    transactionId: TransactionId
+  ): ClosePaymentRequestV2Dto =
+    ClosePaymentRequestV2Dto().apply {
+      paymentTokens =
+        authCompleted.paymentNotices.map { paymentNotice -> paymentNotice.paymentToken.value }
+      outcome = transactionOutcome
+      idPSP = authCompleted.transactionAuthorizationRequestData.pspId
+      paymentMethod = authCompleted.transactionAuthorizationRequestData.paymentTypeCode
+      idBrokerPSP = authCompleted.transactionAuthorizationRequestData.brokerName
+      idChannel = authCompleted.transactionAuthorizationRequestData.pspChannelCode
+      this.transactionId = transactionId.value()
+      totalAmount =
+        EuroUtils.euroCentsToEuro(
+          (authCompleted.transactionAuthorizationRequestData.amount.plus(
+            authCompleted.transactionAuthorizationRequestData.fee)))
+      fee = authCompleted.transactionAuthorizationRequestData.fee.toBigDecimal()
```

Comment:
Use  `EuroUtils.euroCentsToEuro` utils also for fees
As for now you are converting int fee value to BigDecimal without converting from eurocents to euro. So a fee with value 100 will be converted to a BigDecimal valued with 100,00.

Reply:
Fixed with 👉🏻 2b272267f67e0112034f62e459f128c4d2e11b73